### PR TITLE
personal-library-api: jdbc, db

### PR DESCRIPTION
**Pull request by irynarogoza**

### CODE REVIEW

File: src/main/java/com/education/library/config/JdbcConfig.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+
+import javax.sql.DataSource;
+
+
+@Configuration
+@ComponentScan("com.baeldung.jdbc")
```
Comment (delacroix23) [inline src/main/java/com/education/library/config/JdbcConfig.java:14]: remove this line

File: src/main/java/com/education/library/config/JdbcConfig.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+public class JdbcConfig {
+
+    @Bean
+    public DataSource mysqlDataSource() {
+        DriverManagerDataSource dataSource = new DriverManagerDataSource();
+        dataSource.setDriverClassName("com.mysql.jdbc.Driver");
```
Comment (delacroix23) [inline src/main/java/com/education/library/config/JdbcConfig.java:20]: replace driver class name as it deprecated on com.mysql.cj.jdbc.Driver (see application log)

File: src/main/java/com/education/library/config/RestTemplateConfig.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+    @Bean
+    public RestTemplate restTemplate() {
+        return new RestTemplateBuilder()
+                .setConnectTimeout(Duration.ofMillis(30000))
+                .setReadTimeout(Duration.ofMillis(30000))
+                .build();
+    }
+
+}
```
Comment (delacroix23) [inline src/main/java/com/education/library/config/RestTemplateConfig.java:21]: add new line) optional

File: src/main/java/com/education/library/dto/Contributor.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package com.education.library.dto;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor
+@AllArgsConstructor
+@Data
+public class Contributor {
```
Comment (delacroix23) [inline src/main/java/com/education/library/dto/Contributor.java:10]: better to have descending order but it more up on taste 
@Data
@NoArgsConstructor
@AllArgsConstructor

File: src/main/java/com/education/library/dto/IsbnDetails.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+import lombok.NoArgsConstructor;
+
+import java.util.List;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class IsbnDetails {
+
+    private Integer number_of_pages;
```
Comment (delacroix23) [inline src/main/java/com/education/library/dto/IsbnDetails.java:15]: as for training in complex dto construction this class is ok, but main idea in such tasks is to get only necessary fields which will be used, hence all redundant should be removed as they increasing complexity without value for logic

File: src/main/java/com/education/library/service/OpenLibraryService.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Map;
+
+@Service
+public class OpenLibraryService {
+
+    private static final String ISBN_ENDPOINT = "https://openlibrary.org/api/books";
```
Comment (delacroix23) [inline src/main/java/com/education/library/service/OpenLibraryService.java:18]: This value of variable should be moved to application.properties and injected as value

File: src/main/java/com/education/library/service/OpenLibraryService.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        String isbnKey = String.format("ISBN:%s", isbnCode);
+        return response.getBody().getOrDefault(isbnKey, new Isbn());
+
+    }
+
+    private String getUrl(String isbnCode) {
+        return ISBN_ENDPOINT + "?bibkeys=ISBN:" + isbnCode + "&jscmd=details&format=json";
+    }
+
+    //https://openlibrary.org/api/books?bibkeys=ISBN:9780980200447&jscmd=details&format=json
```
Comment (delacroix23) [inline src/main/java/com/education/library/service/OpenLibraryService.java:45]: This line should be removed, no any non informative comments in the code)))

File: src/main/java/com/education/library/service/OpenLibraryService.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+            return null;
+        }
+
+        String isbnKey = String.format("ISBN:%s", isbnCode);
+        return response.getBody().getOrDefault(isbnKey, new Isbn());
+
+    }
+
+    private String getUrl(String isbnCode) {
+        return ISBN_ENDPOINT + "?bibkeys=ISBN:" + isbnCode + "&jscmd=details&format=json";
```
Comment (delacroix23) [inline src/main/java/com/education/library/service/OpenLibraryService.java:42]: add logging of call on debug level

**Review by delacroix23 — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 9]
```diff
@@ -41,6 +43,21 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-jdbc</artifactId>
+            <version>${spring.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+            <version>8.0.21</version>
```
[the remark this answers is not in the document]
Comment (irynarogoza) [reply pom.xml:59]: fixed

File: pom.xml [hunk 10]
```diff
@@ -5,14 +5,15 @@
     <groupId>com.delacroix</groupId>
     <artifactId>personal-library-api</artifactId>
     <version>1.0.0</version>
-    <packaging>war</packaging>
+    <!--<packaging>war</packaging>-->
```
[the remark this answers is not in the document]
Comment (irynarogoza) [reply pom.xml:8]: fixed

File: pom.xml [hunk 11]
```diff
@@ -90,6 +107,49 @@
                     <target>${java.version}</target>
                 </configuration>
             </plugin>
+
+            <plugin>
+                <groupId>org.springframework.boot</groupId>
+                <artifactId>spring-boot-maven-plugin</artifactId>
```
[the remark this answers is not in the document]
Comment (irynarogoza) [reply pom.xml:113]: fixed

File: src/main/java/com/education/library/config/JdbcConfig.java (hunk 1 above)
Comment (irynarogoza) [reply src/main/java/com/education/library/config/JdbcConfig.java:14]: fixed

File: src/main/java/com/education/library/config/JdbcConfig.java (hunk 2 above)
Comment (irynarogoza) [reply src/main/java/com/education/library/config/JdbcConfig.java:20]: fixed

File: src/main/java/com/education/library/dto/Contributor.java (hunk 4 above)
Comment (irynarogoza) [reply src/main/java/com/education/library/dto/Contributor.java:10]: fixed

File: src/main/java/com/education/library/dto/IsbnDetails.java (hunk 5 above)
Comment (irynarogoza) [reply src/main/java/com/education/library/dto/IsbnDetails.java:15]: fixed

File: src/main/java/com/education/library/service/OpenLibraryService.java (hunk 7 above)
Comment (irynarogoza) [reply src/main/java/com/education/library/service/OpenLibraryService.java:45]: fixed

File: src/main/java/com/education/library/service/OpenLibraryService.java (hunk 6 above)
Comment (irynarogoza) [reply src/main/java/com/education/library/service/OpenLibraryService.java:18]: fixed

File: src/main/java/com/education/library/dto/Isbn.java [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class Isbn {
+
+    private String info_url;
```
Comment (delacroix23) [inline src/main/java/com/education/library/dto/Isbn.java:12]: Camel case style for field names

File: src/main/java/com/education/library/dto/IsbnDetails.java [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+
+    @JsonProperty("number_of_pages")
+    private Integer numberOfPages;
+
+    @JsonProperty("table_of_contents")
+    private List<TableContent> tableOfContents;
+
+    private List<Contributor> contributors;
+
+    private List<Integer> isbn_10;
```
Comment (delacroix23) [inline src/main/java/com/education/library/dto/IsbnDetails.java:23]: Camel case

File: src/main/java/com/education/library/service/OpenLibraryService.java [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+
+import java.util.Map;
+
+@Slf4j
+@Service
+@PropertySource("classpath:/propertyfiles/application.properties")
+public class OpenLibraryService {
+
+    @Value("${ISBN_ENDPOINT}")
+    private String isbn_url;
```
Comment (delacroix23) [inline src/main/java/com/education/library/service/OpenLibraryService.java:24]: By code convention we use for static variable **UPPER_CASE_WITH_UNDERSCORE**, for all other fields and variables **camelCaseStyle**

File: src/main/resources/propertyfiles/application.properties [hunk 15]
```diff
@@ -0,0 +1 @@
+ISBN_ENDPOINT=https://openlibrary.org/api/books
```
Comment (delacroix23) [inline src/main/resources/propertyfiles/application.properties:1]: it is not common way to define property with **UPPER_CASE_WITH_UNDERSCORE** please see example in article which you have read https://dev-gang.ru/article/spring-value-annotacija-v08ja8ki2x/

**Review by delacroix23 — CHANGES_REQUESTED** [review]

File: src/main/java/com/education/library/dto/IsbnDetails.java (hunk 13 above)
Comment (irynarogoza) [reply src/main/java/com/education/library/dto/IsbnDetails.java:23]: fixed

File: src/main/java/com/education/library/service/OpenLibraryService.java (hunk 8 above)
Comment (irynarogoza) [reply src/main/java/com/education/library/service/OpenLibraryService.java]: fixed

File: src/main/java/com/education/library/dto/Isbn.java (hunk 12 above)
Comment (irynarogoza) [reply src/main/java/com/education/library/dto/Isbn.java]: fixed

File: src/main/java/com/education/library/service/OpenLibraryService.java (hunk 14 above)
Comment (irynarogoza) [reply src/main/java/com/education/library/service/OpenLibraryService.java]: fixed

File: src/main/resources/propertyfiles/application.properties (hunk 15 above)
Comment (irynarogoza) [reply src/main/resources/propertyfiles/application.properties]: fixed

File: src/main/java/com/education/library/dto/IsbnDetails.java [hunk 16]
```diff
@@ -0,0 +1,102 @@
...
+    private CreatedInfo created;
+
+    private List<String> lccn;
+
+    private String notes;
+
+    private Identifier identifiers;
+
+    @JsonProperty("isbn_13")
+    private List<String> isbn13;
```
Comment (delacroix23) [inline src/main/java/com/education/library/dto/IsbnDetails.java:90]: just to let you know here the name of variable could not match original key which you stated in @JsonProperty

File: src/main/java/com/education/library/dto/IsbnDetails.java (hunk 13 above)
Comment (delacroix23) [reply src/main/java/com/education/library/dto/IsbnDetails.java:23]: line 23

**Review by delacroix23 — CHANGES_REQUESTED** [review]

**Review by delacroix23 — APPROVED** [review]

**Pull request merged**